pkgs/test/nixpkgs-check-by-name/src/eval.rs

```diff
@@ -103,74 +118,88 @@ pub fn check_values(
         anyhow::bail!("Failed to run command {command:?}");
     }
     // Parse the resulting JSON value
-    let actual_files: HashMap<String, AttributeInfo> = serde_json::from_slice(&result.stdout)
+    let attributes: Vec<(String, ByNameAttribute)> = serde_json::from_slice(&result.stdout)
         .context(format!(
             "Failed to deserialise {}",
             String::from_utf8_lossy(&result.stdout)
         ))?;
```

Comment:
Here and throughout the crate: use `anyhow::Context::with_context` instead of `context` to avoid allocating these error strings in the non-error case.

Reply:
Done throughout the codebase in https://github.com/NixOS/nixpkgs/pull/278805/commits/a1db0cdf9bb5e90727eb1c455da448fb21bab91e :)